### ISSUE TITLE
Lifted sequence coercions

### ISSUE THREAD

**Issue by treeowl** [atzeus/reflectionwithoutremorse#1]
Currently, you use `data Any a where Any :: a b c -> Any a`. This has a potential efficiency problem because of the extra constructor. I'm also not sure if it's entirely safe. As best I can tell, the officially sanctioned way to do this sort of thing is with the somewhat magical `GHC.Prim.Any :: *`, which is designed as a "safe" intermediary for `unsafeCoerce`. To use it, you'd just make a `Seq Any` or similar, and `unsafeCoerce` both on the way in and on the way out. GHC seems to promise not to play games with this type.

``` haskell
newtype FakeSequence a b c = FakeSequence (Seq Any)
```

**Comment by atzeus** [comment]
Where do I use Any? Cheers!

**Comment by atzeus** [comment]
O wait in LiftSequence :)

**Comment by atzeus** [comment]
Yep this is definitely an improvement! I'll change it. Thanks!

**Comment by treeowl** [comment]
I guess you'll want a kind annotation there. Probably something like

``` haskell
newtype FakeSequence (a :: k -> k -> *) (b::k) (c::k) = FakeSequence (Seq Any)
```

**Comment by treeowl** [comment]
Any progress on this? Should I open a pull request?

**Comment by atzeus** [comment]
Fixed by #2ab000be1ddc49a2ec36afaaa58485df56aef654

**Comment by treeowl** [comment]
I just looked at this again, and was surprised to see `unsafeCoerce#` rather than the "safer" `unsafeCoerce`. Why is that primitive needed here?
